build/tools/Automation/DotNet/DotNetAutomator.cs

```diff
@@ -58,8 +58,22 @@ public async Task RunAsync()
                 this.oryxSdkStorageBaseUrl = Constants.OryxSdkStorageBaseUrl;
             }
 
-            this.oryxDotNetSdkVersions = await this.httpService.GetOryxSdkVersionsAsync(
-                this.oryxSdkStorageBaseUrl + DotNetConstants.DotNetSuffixUrl);
+            string sdkVersionsUrl = this.oryxSdkStorageBaseUrl + DotNetConstants.DotNetSuffixUrl;
+
+            // A SAS token is required for the staging account.
+            if (this.oryxSdkStorageBaseUrl == Constants.OryxSdkStagingStorageBaseUrl)
+            {
+                string sasToken = Environment.GetEnvironmentVariable(Constants.OryxSdkStagingPrivateSasTokenEnvVar);
+                if (string.IsNullOrEmpty(sasToken))
+                {
+                    throw new ArgumentException($"The environment variable {Constants.OryxSdkStagingPrivateSasTokenEnvVar} " +
+                        $"must be provided in order to access {Constants.OryxSdkStagingStorageBaseUrl}");
+                }
+
+                sdkVersionsUrl += "&" + sasToken;
+            }
+
+            this.oryxDotNetSdkVersions = await this.httpService.GetOryxSdkVersionsAsync(sdkVersionsUrl);
```

Comment:
nit: if we have a `Common` project (or some equivalent set of common files shared amongst the different platforms), it would be nice to move this block of logic (from lines 55 to 76) out into its own method that's shared amongst the platforms, and all they need to do is provide the base URL to the method call.

Reply:
+1

Reply:
Yeah let me get this one in first and I can add that separately. Since the pipelines seem to have noticeable transient intermittent errors.

Reply:
https://github.com/microsoft/Oryx/pull/2092